recipes/toolchain/activate.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+if [ "$(uname)" == "Darwin" ]
+then
+    # for Mac OSX
+    export CC=clang
+    export CXX=clang++
+    export MACOSX_VERSION_MIN="10.7"
+    export MACOSX_DEPLOYMENT_TARGET="${MACOSX_VERSION_MIN}"
+    export CXXFLAGS="${CXXFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export CXXFLAGS="${CXXFLAGS} -stdlib=libc++"
+    export LDFLAGS="${LDFLAGS} -mmacosx-version-min=${MACOSX_VERSION_MIN}"
+    export LDFLAGS="${LDFLAGS} -stdlib=libc++ -lc++"
+    export LINKFLAGS="${LDFLAGS}"
+elif [ "$(uname)" == "Linux" ]
+then
+    # for Linux
+    export CC=gcc
+    export CXX=g++
+    export CXXFLAGS="${CXXFLAGS}"
```

Comment:
Might want to add ` -DBOOST_MATH_DISABLE_FLOAT128` to disable Boost float128 support. As we don't provide `libquadmath`, this seems like a sensible change. This is inspired by this Boost ticket ( https://svn.boost.org/trac/boost/ticket/9240 ). It is starting to look like the kind of thing we are going to need quite a bit when using Boost on Linux.